### PR TITLE
Improve error handling on Signup form

### PR DESCRIPTION
**Pull request by ssvmvss**
Improve error handling on Signup form:
- Display different error messages per field.
- Clear each error message when the user starts typing on each field.

### CODE REVIEW

File: src/routes/signup/views/components/SignupForm.jsx [hunk 1]
```diff
@@ -77,78 +77,96 @@ const ShowPasswordOption = styled.div`
...
 
+/**
+ *
+ */
+const SignupForm = ({ product, plan, redirect, errors = {} }) => {
+
+  const {
+    company: companyError,
+    email: emailError,
+    password: passwordError
```
Comment (esclapes) [inline src/routes/signup/views/components/SignupForm.jsx]: @ssvmvss This is a nice pattern :heart_eyes: I will make sure to send you the errors in this way with the new `command` version 

What should we do with any other errors? Maybe we could print them at the top of the form. We could use destructuring to make it very generic and catch all other possible errors

```
const {
    company: companyError,
    email: emailError,
    password: passwordError,
   ...otherErrors
} = errors
```

Comment (ssvmvss) [reply src/routes/signup/views/components/SignupForm.jsx]: Good question, Edu! I've pushed a little fix to display general errors on top. I have some doubts about the approach so I'll reach out to you soon to see if we can check some things together :)

File: src/routes/signup/views/components/SignupForm.jsx [hunk 2]
```diff
@@ -1,153 +1,93 @@
...
+        <Input
+          id="email"
+          type="email"
+          name="email"
+          label="Work Email"
+          defaultValue=""
+          error={emailError}
+          className={emailError && 'error'}
+        />
+        <InputPasswordSet
```
Comment (esclapes) [inline src/routes/signup/views/components/SignupForm.jsx]: Nice new component :100:

File: src/routes/signup/views/components/SignupForm.jsx [hunk 3]
```diff
@@ -1,153 +1,93 @@
...
+          hidden
+        />
+        <Button id="signup-form-submit" type="submit" disabled>
+          Continue
+        </Button>
+        <LinksGroup justify="space-between">
+          <p>I agree to <Link href="https://buffer.com/terms" blue>Buffer's Terms of Service</Link></p>
+          <Link href={`/login?redirect=${redirect}`}>Already have an account?</Link>
+        </LinksGroup>
+        {redirect && <input type="hidden" name="redirect" value={redirect} />}
```
Comment (esclapes) [inline src/routes/signup/views/components/SignupForm.jsx]: :thinking: I am not sure if we are going to need the redirect here... With the current flow, we have a clear destination after signup up, i.e. `reply.buffer.com/onboarding`.

I may be forgetting some use case, but after merging #54 we would be ignoring this parameter. Let me know if you think we should add it back :)

Comment (ssvmvss) [reply src/routes/signup/views/components/SignupForm.jsx]: Hey Edu! Thank you for mentioning this! The redirect parameter is just there as a quick and easy way to build the Login URL. I have a pending To-Do to delete that and actually use one of our "getXURL()". What I want to do is something similar like you did here: https://github.com/bufferapp/buffer-login/pull/55 but with an util function that I saw you already added here: https://github.com/bufferapp/buffer-login/pull/54

Once #54 is merged I'll create another PR to clean this part using the `getProductUrl()` util.

How does that sound?

**Comment by ssvmvss** [comment]
Closing this PR. We included this changes on this other PR instead: https://github.com/bufferapp/buffer-login/pull/60

**Pull request closed without merging**